src/spikeinterface/curation/curation_format.py

```diff
@@ -134,30 +138,206 @@ def curation_label_to_dataframe(curation_dict):
+      * for merged group, when exclusive=True, if all have the same label then this label is applied
+      * for merged group, when exclusive=False, if one unit has the label then the new one have also it
+    """
+
+    # Please note that manual_labels is done on the unit_ids before the merge!!!
+    manual_labels = curation_label_to_vectors(curation_dict)
+
+    # apply on non merged
+    for key, values in manual_labels.items():
+        all_values = np.zeros(sorting.unit_ids.size, dtype=values.dtype)
+        for unit_ind, unit_id in enumerate(sorting.unit_ids):
+            if unit_id not in new_unit_ids:
+                ind = curation_dict["unit_ids"].index(unit_id)
+                all_values[unit_ind] = values[ind]
+        sorting.set_property(key, all_values)
+    
+    for new_unit_id, old_group_ids in zip(new_unit_ids, curation_dict["merge_unit_groups"]):
+        for label_key, label_def in curation_dict["label_definitions"].items():
+            if label_def["exclusive"]:
+                group_values = []
+                for unit_id in old_group_ids:
+                    ind = curation_dict["unit_ids"].index(unit_id)
+                    value = manual_labels[label_key][ind]
+                    if value != '':
+                        group_values.append(value)
+                if len(set(group_values)) == 1:
+                    # all group has the same label or empty
+                    sorting.set_property(key, values=group_values, ids=[new_unit_id])
+            else:
+
+                for key in label_def["label_options"]:
+                    group_values = []
+                    for unit_id in old_group_ids:
+                        ind = curation_dict["unit_ids"].index(unit_id)
+                        value = manual_labels[key][ind]
+                        group_values.append(value)
+                    new_value = np.any(group_values)
+                    sorting.set_property(key, values=[new_value], ids=[new_unit_id])
+
+
+
+def apply_curation(sorting_or_analyzer, curation_dict, censor_ms=None, new_id_strategy="append",
+                   merging_mode="soft", sparsity_overlap=0.75, verbose=False,
+                   **job_kwargs):
+    """
+    Apply curation dict to a Sorting or an SortingAnalyzer.
+
+    Steps are done this order:
+      1. Apply removal using curation_dict["removed_units"]
+      2. Apply merges using curation_dict["merge_unit_groups"]
+      3. Set labels using curation_dict["manual_labels"]
+
+    A new Sorting or SortingAnalyzer (in memory) is returned.
+    The user (an adult) has the responsability to save it somewhere (or not).
+
+    Parameters
+    ----------
+    sorting_or_analyzer : Sorting | SortingAnalyzer
+        The Sorting object to apply merges.
+    curation_dict : dict
+        The curation dict.
+    censor_ms: float | None, default: None
+        When applying the merges, should be discard consecutive spikes violating a given refractory per
```

Comment:
```suggestion
        When applying the merges, any consecutive spikes within the `censor_ms` are removed. This can be thought of
        as the desired refractory period. If `censor_ms=None`, no spikes are discarded. 
```